src/Layout/Page/Page.component.tsx

```diff
@@ -2,50 +2,215 @@
+import classnames from 'classnames';
 // COMPONENTS
-import Header from '../Header/Header.component';
-import Footer from '../Footer/Footer.component';
 import { NormalizeCSS, GlobalCSS } from '../../theme/GlobalStyles';
+import { CustomAdaptor, StandardBreakpoints } from '../../Grid/Adaptor';
 
-const StyledPageElement = styled.section`
+// Constants
+// ------------------------------------------------------------------------------------------------------------------
+export const DEFAULT_LAYOUT_WIDTH = '100%';
+export const DEFAULT_CONTENT_WIDTH = '71.25rem';
+export const DEFAULT_SIDEBAR_WIDTH = '13.75rem';
+export const TRANSPARENT = 'transparent';
+export const DEFAULT_HEADER_HEIGHT = '1.5rem';
+export const DEFAULT_FOOTER_HEIGHT = '20.375rem';
+const RIGHT = 'right';
+
+// Defaults
+// ------------------------------------------------------------------------------------------------------------------
+const PageDefaultProps = {
+    headerHeight: DEFAULT_HEADER_HEIGHT,
+    footerHeight: DEFAULT_FOOTER_HEIGHT,
+};
+
+const DefaultLayoutDefaultProps = {
+    layoutWidth: DEFAULT_LAYOUT_WIDTH,
+    contentWidth: DEFAULT_CONTENT_WIDTH,
+    layoutBackgroundColor: TRANSPARENT,
+    contentBackgroundColor: TRANSPARENT,
+};
+
+const SidebarLayoutDefaultProps = {
+    sidebarWidth: DEFAULT_SIDEBAR_WIDTH,
+    layoutWidth: DEFAULT_CONTENT_WIDTH,
+    contentWidth: DEFAULT_CONTENT_WIDTH,
+    layoutBackgroundColor: TRANSPARENT,
+    contentBackgroundColor: TRANSPARENT,
+};
+
+// Types
+// ------------------------------------------------------------------------------------------------------------------
+type SidebarAlignment = 'left' | 'right';
+
+// Styled Components
+// ------------------------------------------------------------------------------------------------------------------
+const StyledPage = styled.section`
     display: flex;
     flex-direction: column;
     height: 100vh;
-    max-width: 100vw;
-
-    header {
-        flex: 0 0 auto;
-        max-width: 100vw;
-    }
+    max-width: 100%;
+`;
 
-    footer {
-        flex: 0 0 auto;
-        max-width: 100vw;
-    }
+const StyledLayout = styled.section<StyledLayoutProps>`
+    max-width: ${props => props.layoutWidth};
+    background-color: ${props => props.layoutBackgroundColor};
+    margin: 0 auto;
 `;
 
-const StyledBodyElement = styled.section`
-    flex: 1 0 auto;
-    background-color: ${colors.silver.base};
-    max-width: 100vw;
-    align-self: stretch;
-    padding: ${sizes.padding.lg} ${sizes.padding.xl};
+const StyledContent = styled.div<StyledContentProps>`
+    max-width: ${props => props.contentWidth};
+    background-color: ${props => props.contentBackgroundColor};
+    margin: 0 auto;
 `;
 
+// Interfaces
+// ------------------------------------------------------------------------------------------------------------------
 interface PageProps {
+    /** The header of the website. This can be a component. */
+    header?: any;
```

Comment:
I don't know the answer to this question, but is there a way we can enforce only the right element types AND use `children`? My only hack to finding out how to do this is to type-cast the props (which works with what you have here), but I'd like to be able to pass all these elements in as children and have them all work 'automatically'.

Reply:
Using props is more standard/natural way of approaching the problem, and I think trying to not be too clever on this is the best way to go. Still, the idea of sniffing out a child's type and applying it correctly is an interesting notion. Definitely would like to talk about this because it's intriguing.

Reply:
Maybe we add a v2 ticket. This can be outside of MVP, but still worth considering, right?